var/spack/repos/builtin/packages/pism/package.py

```diff
@@ -80,9 +81,9 @@ class Pism(CMakePackage):
     depends_on("gsl")
     depends_on("mpi")
     depends_on("netcdf-c")  # Only the C interface is used, no netcdf-cxx4
-    depends_on("petsc")
+    depends_on("petsc@3.20+scalapack+mumps+ptscotch")
```

Comment:
PISM is not compatible with `petsc+complex`, but other than that just `petsc` should be sufficient.